tools/bag_processing/scripts/Makefile.rosbag_fix_all

```diff
@@ -28,42 +28,63 @@
 
 ROS_VERSION=$(shell rosversion -d)
 
+# Find script executables
 ifeq (,${REWRITE_TYPES})
   REWRITE_TYPES = $(shell catkin_find --first-only bag_processing scripts/rosbag_rewrite_types.py)
 endif
 ifeq (,${ROSBAG_VERIFY})
   ROSBAG_VERIFY = $(shell catkin_find --first-only bag_processing scripts/rosbag_verify.py)
 endif
+ifeq (,${ROSBAG_DEBAYER})
+  ROSBAG_DEBAYER = $(shell catkin_find --first-only bag_processing scripts/rosbag_debayer.py)
+endif
+ifeq (,${ROSBAG_SPLIT_DEPTH})
+  ROSBAG_SPLIT_DEPTH = $(shell catkin_find --first-only pico_driver scripts/pico_split_extended.py)
+endif
+ifeq (,${ROSBAG_FILTER})
+  ROSBAG_FILTER = $(shell catkin_find --first-only bag_processing scripts/rosbag_topic_filter.py)
+endif
+
+# Find rosbag migration definitions
 ifeq (,${FF_MSGS_BMR})
   FF_MSGS_BMR = $(shell catkin_find --first-only ff_msgs bmr)
 endif
 BMR = $(sort $(wildcard ${FF_MSGS_BMR}/*.bmr) $(wildcard ${FF_MSGS_BMR}/${ROS_VERSION}/*.bmr))
-REWRITE_RULES = -r ${FF_MSGS_BMR}/rosbag_rewrite_types_rules.json
+REWRITE_TYPES_ARGS += -r ${FF_MSGS_BMR}/rosbag_rewrite_types_rules.json
 
 ifeq (,${ISAAC_MSGS_BMR})
   ISAAC_MSGS_BMR = $(shell catkin_find --first-only isaac_msgs bmr 2> /dev/null)
 endif
 ifneq (,${ISAAC_MSGS_BMR})
   # If isaac_msgs is available, apply extra rules. Shouldn't hurt non-ISAAC bags.
   BMR += $(sort $(wildcard ${ISAAC_MSGS_BMR}/*.bmr))
-  REWRITE_RULES += -r ${ISAAC_MSGS_BMR}/rosbag_rewrite_types_rules.json
+  REWRITE_TYPES_ARGS += -r ${ISAAC_MSGS_BMR}/rosbag_rewrite_types_rules.json
 endif
 
+# Fix rosjava scicam bug
 %.rewrite_types.bag: %.bag
-	${REWRITE_TYPES} -v ${REWRITE_RULES} $< -o $@
+	${REWRITE_TYPES} -v ${REWRITE_TYPES_ARGS} $< -o $@
 
-%.fix_all_pre_check.bag: %.rewrite_types.bag
+# Fix old message definitions
+%.migrate_old.bag: %.rewrite_types.bag
 	rosbag fix $< $@ ${BMR}
+	rosbag check $@  # test: post-migration consistency
+	# compare with original bag to make sure data is there
+	${ROSBAG_VERIFY} ${ROSBAG_VERIFY_ARGS} $@  $(@:.migrate_old.bag=.bag)
 	rm $<  # explicitly delete right away to save disk space
 
-# The fix_all_check targets are .PHONY but too much trouble and not
-# necessary to declare that.
+# Debayer the nav cam
+%.debayer.bag: %.migrate_old.bag
+	${ROSBAG_DEBAYER} ${ROSBAG_DEBAYER_ARGS} $< -o $@
+	rm $<  # explicitly delete right away to save disk space
 
-fix_all_check1-%: %.fix_all_pre_check.bag
-	rosbag check $<
+# Decode the haz cam into point cloud + image
+%.depth_split.bag: %.debayer.bag
+	${ROSBAG_SPLIT_DEPTH} ${ROSBAG_SPLIT_DEPTH_ARGS} $< -o $@
```

Comment:
It's a bummer that these steps will always be run as a pass-through that copies the large bag file, even when the relevant processing features are disabled. Could your `shutil.copyfile()` call be replaced with an O(1) call to `os.rename()` to avoid the copying overhead? If you don't want the subsequent `rm` command to fail, I think you could use something like `[ -e $< ] && rm $<`.